sql/repl_failsafe.cc

```diff
@@ -172,6 +172,8 @@ static my_bool show_slave_hosts_callback(THD *thd, Protocol *protocol)
   mysql_mutex_lock(&thd->LOCK_thd_data);
   String gtid_sent[GTID_MAX_STR_LENGTH];
   String gtid_ack[GTID_MAX_STR_LENGTH];
+  const char *sync_str_opt[]= {"async","semi-sync","semi-sync stalled"};
```

Comment:
i think the status options can be a little improved. `semi-sync stalled` may be somewhat misleading because it may incorrectly imply an error, when the slave may just be doing an initial round of catch up; `semi-sync stale` reads a little less aggressive. Then just `semi-sync` doesn't provide an explicit status, and should provide an opposite to `stale`, e.g. `semi-sync active`. 

I also think `async` would read better as `asynchronous`, but that may be just personal taste :)